### PR TITLE
[IMP] account: make journal entry optional for payments

### PR DESCRIPTION
**Pull request by william-andre**
TLDR
====

A payment doesn't always generate a journal entry anymore, it will only do so if an outstanding account is set.

Purpose
=======

Payment management in Odoo should be treated as an extra-accounting process and should not generate accounting entries automatically. Recording payments is primarily used to prevent duplicate payments of the same invoice and to prepare bank files, such as SEPA Credit Transfers. However, it is common for payments to be canceled, rejected, or deleted due to errors, such as insufficient funds in the client’s account or forgotten file submissions.

Another goal of this simplification is to make bank reconciliation easier. The process involves simply identifying the partner and, if possible, matching a document. If immediate matching isn’t feasible, it’s not a problem; you can associate the bank transaction with the partner’s account and complete the matching later. This approach minimizes errors and greatly simplifies the workflow. Currently, the procedure is more complex, often requiring matching within the outstanding account or directly with the partner, which increases confusion and the risk of mistakes (in cases of payments).

Currently, the accounting entries generated by payments can clutter the customer and vendor balances, making data interpretation more challenging. For example, instead of seeing only the invoice entries related to bank transactions, users often see payment entries, which complicates management.

Dates are also an issue, as we often see the payment date, but ultimately, only the bank date is truly useful.

Implementation
==============

Before the payment used to be a composition based inheritance (a.k.a. `_inherits`) of the journal entry, but the journal entry doesn't always exist anymore, we can't rely on that mechanism anymore.
* Payments can be generated from scratch (the "New" button), in that case, the payment will generate it's journal entry upon validation, if and only is there is an outstanding account set.
* Payments can also be generated from various other sources (i.e. "Register Payment", from Expenses, from the PoS,...). In those cases, the source is responsible to generate the journal entry and link it to the payment. By simplicity, the lines to be created can be provided to the create values. In these case the payment won't go through the draft state.
* This also means that there is no synchronisation anymore between the payment and the journal entry.
* Because there isn't always a journal entry anymore, we needed to add some fields on the payment which might be duplidated on the journal entry if there is one: Sent, Memo, Name and State.
* Now that the payments can not have a journal entry, it is impossible to rely only on the reconciliation mechanism to link the payments and the invoices. New many2many fields `invoice_ids`/`matched_payment_ids` have been added between the two models.

Details
=======

* There is now a "(Un)Block Payment" button on the invoice in order to prevent registering a payment if needed.
* When an invoice is fully reconciled (with a bank transaction), the related payments will be marked as Paid. More and improved heuristics will be added in the future to avoid keeping pending payments as In Process.
* Payments without journal entries don't appear in the paid amount, but information about pending payments will be added in the future on invoice reports and the followup.
* New states have been added on the payment. Posted is renamed in To Process, and 3 completely new states are added: 
  * Paid: there was previously a checkbox Matched, but that was only relevant for reconciliation, which is not possible anymore 
  * Canceled: the user doesn't want to proceed with this payment anymore 
  * Rejected: the payment was rejected by the payment provider (insufficient funds, wrong details,...)
* Internal Transfers have been removed: it should be matched as a writeoff during bank reconciliation instead.

[task-4180163](https://www.odoo.com/odoo/project.task/4180163)

### CODE REVIEW

File: addons/account/models/account_payment.py [hunk 1]
```diff
@@ -869,240 +848,86 @@ def create(self, vals_list):
             # Hack to force a custom balance.
             force_balance_vals_list.append(vals.pop('force_balance', None))
 
-            # Force the move_type to avoid inconsistency with residual 'default_move_type' inside the context.
-            vals['move_type'] = 'entry'
+            # Hack to add a custom line.
```
Comment (qdp-odoo) [inline addons/account/models/account_payment.py:851]: what kind of custom line? there's already 2 other 'custom' lines use cases before... :upside_down_face:

File: addons/account/models/account_payment.py [hunk 2]
```diff
@@ -869,240 +848,86 @@ def create(self, vals_list):
...
-
-        # We need to reset the cached name, since it was recomputed on the delegate account.move model
-        payments.invalidate_recordset(fnames=['name'])
-        return payments.with_env(self.env)  # clear the context
+            if (
+                write_off_line_vals_list[i] is not None
+                or force_balance_vals_list[i] is not None
+                or linecomplete_line_vals_list[i] is not None
+            ):
+                pay._synchronize_to_moves(
```
Comment (qdp-odoo) [inline addons/account/models/account_payment.py:862]: how are you sure you need to sync with a move? maybe the payment didn't create any...

only relying on giving values for the related fields on move_id.something? 

not even sure what you're trying to acheive there, tbh

File: addons/account/models/account_payment.py [hunk 3]
```diff
@@ -869,240 +848,86 @@ def create(self, vals_list):
...
+                    force_balance=force_balance_vals_list[i],
+                    line_ids=linecomplete_line_vals_list[i],
+                )
+                # propagate the related fields to the move as it is being created after the payment
+                if move_vals := {
+                    fname: value
+                    for fname, value in vals.items()
+                    if self._fields[fname].related and (self._fields[fname].related or '').split('.')[0] == 'move_id'
+                }:
+                    pay.move_id.write(move_vals)
```
Comment (qdp-odoo) [inline addons/account/models/account_payment.py:873]: ok, after having read _synchronize_to_moves, I better understand but I don't like it much: can't we move that related fields propagation inside synchronize_to_moves, and do it directly during the move creation ? I don't see any reason why to keep that in here :thinking: 


what do you think?

File: addons/account/models/account_payment.py [hunk 4]
```diff
@@ -8,36 +8,58 @@
...
+        string='Journal Entry',
         index=True,
+        copy=False,
         check_company=True)
+    invoice_ids = fields.Many2many(
+        string="Invoices",
+        comodel_name='account.move',
+        relation='account_move__account_payment',
+        column1='payment_id',
+        column2='invoice_id',
```
[the remark this answers is not in the document]
Comment (qdp-odoo) [reply addons/account/models/account_payment.py:30]: why do we have invoice_ids and reconciled_invoice_ids ??

**Review by qdp-odoo — CHANGES_REQUESTED** [review]
partial review PART 1

File: addons/account/models/account_move.py [hunk 5]
```diff
@@ -50,6 +50,7 @@
         ('paid', 'Paid'),
         ('partial', 'Partially Paid'),
         ('reversed', 'Reversed'),
+        ('blocked', 'Blocked'),
```
[the remark this answers is not in the document]
Comment (william-andre) [reply addons/account/models/account_move.py:53]: Why do we need that field?

File: addons/account/models/account_payment.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (william-andre) [reply addons/account/models/account_payment.py:30]: `reconciled_invoice_ids` is when there is a reconciliation in the moves lines, with a matching number; meaning that the residual amount has been decreased

File: addons/account/models/account_payment.py [hunk 6]
```diff
@@ -178,7 +192,8 @@ class AccountPayment(models.Model):
     amount_company_currency_signed = fields.Monetary(
         currency_field='company_currency_id', compute='_compute_amount_company_currency_signed', store=True)
     # used to get and display duplicate move warning if partner, amount and date match existing payments
-    duplicate_move_ids = fields.Many2many(comodel_name='account.move', compute='_compute_duplicate_move_ids')
+    duplicate_payment_ids = fields.Many2many(comodel_name='account.payment', compute='_compute_duplicate_move_ids')
+    attachment_ids = fields.One2many('ir.attachment', 'res_id', string='Attachments')
```
[the remark this answers is not in the document]
Comment (william-andre) [reply addons/account/models/account_payment.py:196]: I guess so, I can't remember. It doesn't hurt to have it anyway

File: addons/account/models/account_payment.py [hunk 7]
```diff
@@ -359,6 +367,45 @@ def _prepare_move_line_default_vals(self, write_off_line_vals=None, force_balanc
...
+                ('type', 'in', ['bank', 'cash', 'credit']),
+            ], limit=1)
+
+    @api.depends('journal_id')
+    def _compute_company_id(self):
+        for payment in self:
+            if payment.journal_id.company_id not in payment.company_id.parent_ids:
+                payment.company_id = (payment.journal_id.company_id or self.env.company)._accessible_branches()[:1]
+
+    @api.depends('invoice_ids.payment_state', 'is_matched')
```
[the remark this answers is not in the document]
Comment (william-andre) [reply addons/account/models/account_payment.py:397]: Indeed, artifact from previous implem. Removing it :+1:

File: addons/account/models/account_payment.py (hunk 1 above)
Comment (william-andre) [reply addons/account/models/account_payment.py:851]: The PoS for instance creates an entirely custom payment as well... Not just writeoff lines.
Just trying to not break anything, we will probably need to re-add some sync mechanism in the near future anyway

File: addons/account/models/account_payment.py (hunk 2 above)
Comment (william-andre) [reply addons/account/models/account_payment.py:862]: The method name is probably confusing, it's not syncing, it's creating the move

Comment (william-andre) [reply addons/account/models/account_payment.py:862]: Renaming to `_generate_journal_entry`

File: addons/account/models/account_payment.py (hunk 3 above)
Comment (william-andre) [reply addons/account/models/account_payment.py:873]: You have in mind that there is still a sync being done.
While that would be ideal, some influential people decided that it was not necessary.
We will probably come back on that in the near future...

File: addons/account/models/chart_template.py [hunk 8]
```diff
@@ -731,20 +731,6 @@ def _setup_utility_bank_accounts(self, template_code, company, template_data):
...
-                'account_type': 'asset_current',
-                'reconcile': True,
-            },
-            'account_journal_payment_credit_account_id': {
-                'name': _("Outstanding Payments"),
-                'prefix': bank_prefix,
-                'code_digits': code_digits,
-                'account_type': 'asset_current',
-                'reconcile': True,
-            },
```
[the remark this answers is not in the document]
Comment (william-andre) [reply addons/account/models/chart_template.py:747]: The account is still created if it is in the data of the l10n, this was only a fallback.
But this is also only setting the value of a field that doesn't exist anymore

File: addons/account/views/account_payment_view.xml [hunk 9]
```diff
@@ -13,9 +13,9 @@
                         <button name="action_post" type="object" string="Confirm"/>
                     </header>
                     <field name="company_currency_id" column_invisible="True"/>
-                    <field name="suitable_journal_ids" column_invisible="True"/>
+                    <!-- <field name="suitable_journal_ids" column_invisible="True"/> -->
```
Comment (qdp-odoo) [inline addons/account/views/account_payment_view.xml:16]: to be cleaned afterward...

File: addons/account/models/account_move.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (qdp-odoo) [reply addons/account/models/account_move.py:53]: we used to use that in followup reports: thicking it to True would depict that a receivable line was in litigation, and so you don't want to take it into account for your followups (it is ignored). Probably a duplicated feature with yours, maybe already removed by Habib in his branch...

File: addons/event_booth_sale/tests/test_event_booth_sale.py [hunk 10]
```diff
@@ -207,4 +207,6 @@ def test_event_booth_with_invoice(self):
             f"Invoice payment is in '{invoice.payment_state}' status while it should be '{in_payment_state}'.")
 
         self.assertEqual(booth.state, 'unavailable')
+        self.assertFalse(booth.is_paid)  # it is still pending reconciliation
+        invoice._invoice_paid_hook()
```
Comment (qdp-odoo) [inline addons/event_booth_sale/tests/test_event_booth_sale.py:211]: It would be better to process the payment, and have the booth state updated automatically :point_up:

File: addons/hr_expense/models/account_payment.py [hunk 11]
```diff
@@ -1,13 +1,25 @@
...
+    expense_sheet_id = fields.Many2one(related='move_id.expense_sheet_id')
+
+    def write(self, vals):
+        trigger_fields = {
+            'date', 'amount', 'payment_type', 'partner_type', 'payment_reference',
+            'currency_id', 'partner_id', 'destination_account_id', 'partner_bank_id', 'journal_id'
+            'ref', 'expense_sheet_id', 'payment_method_line_id'
+        }
+        if self.expense_sheet_id and any(field_name in trigger_fields for field_name in vals):
+            raise UserError(_("You cannot change the expense report linked to a payment."))
```
Comment (qdp-odoo) [inline addons/hr_expense/models/account_payment.py:20]: error message doesn't make any sense: we're modifying a payment. What was wrong with the previous one?

File: addons/hr_expense/views/res_config_settings_views.xml [hunk 12]
```diff
@@ -43,6 +43,10 @@
                                      string="Employee Expense Journal">
                                 <field name="expense_journal_id"/>
                             </setting>
+                            <setting company_dependent="1" help="Default outstanding account for expenses paid by employees."
+                                     string="Employee Expense Oustanding Account">
+                                <field name="expense_outstanding_account_id"/>
```
Comment (qdp-odoo) [inline addons/hr_expense/views/res_config_settings_views.xml:48]: tbh, it feels like unfinished business: we removed the outstanding account but we still need it for expenses... :see_no_evil: and we won't have it automatically filled with the CoA generation :shrug:

File: addons/l10n_latam_check/models/account_chart_template.py [hunk 13]
```diff
@@ -30,11 +33,26 @@ def _get_latam_check_account_journal(self, template_code):
...
                     ],
                     'inbound_payment_method_line_ids': [
                         Command.create({'payment_method_id': self.env.ref('l10n_latam_check.account_payment_method_new_third_party_checks').id}),
                         Command.create({'payment_method_id': self.env.ref('l10n_latam_check.account_payment_method_in_third_party_checks').id}),
                     ],
                 },
             }
+
+    @template(model='account.account')
+    def _get_latam_check_outstanding_account_account(self, template_code):
```
Comment (qdp-odoo) [inline addons/l10n_latam_check/models/account_chart_template.py:49]: hoooooo that's clever :) I suppose you can ignore one of my previous comment about the lost of the outstanding accounts in the chart templates :-)

nice one nice one :+1:

File: addons/l10n_latam_check/models/account_chart_template.py [hunk 14]
```diff
@@ -30,11 +33,26 @@ def _get_latam_check_account_journal(self, template_code):
...
+        if self.env.company.country_id.code in self._get_third_party_checks_country_codes():
+            return {
+                'outstanding_check_out': {
+                    'name': "Outstanding Check Out",
```
Comment (qdp-odoo) [inline addons/l10n_latam_check/models/account_chart_template.py:53]:
```suggestion
                    'name': _("Outstanding Check Out"),
```

**Review by qdp-odoo — CHANGES_REQUESTED** [review]
partial review PART 2

File: addons/hr_expense/models/account_payment.py (hunk 11 above)
Comment (william-andre) [reply addons/hr_expense/models/account_payment.py:20]: It's the same behavior as before, but it used to be done in the sync

File: addons/hr_expense/views/res_config_settings_views.xml (hunk 12 above)
Comment (william-andre) [reply addons/hr_expense/views/res_config_settings_views.xml:48]: It was discussed with TSB, there is no other choice because we always need to generate an entry for company expenses, but we don't want to depend on the configuration of the payment method

**Review by qdp-odoo — CHANGES_REQUESTED** [review]
partial review part 3/3. 

Very good job, William :) Special mention for the commit message, very detailed and motivated, even if the block "Implementation" would need a bit of a rework I think:

> Before the payment used to be a composition based inheritance (a.k.a.
> `_inherits`) of the journal entry, but now the journal entry doesn't always
> exist anymore:
> * Payments can be generated from scratch (the "New" button), in that case,
> the payment will generate its journal entry only upon validation, if and only if there's an outstanding account set.
> * Payments can also be generated from various other sources (i.e.
> "Register Payment", from Expenses, from the PoS,...). In those cases,
> the source is responsible to generate the journal entry and link it to
> the payment. For the sake of simplicity, the lines to be created can be provided to
> the create values. In these cases the payment won't go through the draft
> state.
> * This also means that there is no synchronization anymore between the
> payment and the journal entry. 
> * Because there isn't always a journal entry anymore, we needed to add
> some fields on the payment which might be duplicated on the journal
> entry if there is one: Sent, Memo, Name, State...
> * Now that since the payments can have no journal entry, it is impossible to
> rely only on the reconciliation mechanism to link the payments and the
> invoices. A new many2many field 'invoice_ids' has been added between the two models.

File: addons/account/models/chart_template.py (hunk 8 above)
[the remark this answers is not in the document]
Comment (qdp-odoo) [reply addons/account/models/chart_template.py:747]: saw that :) :+1:

File: addons/account/models/account_payment.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (qdp-odoo) [reply addons/account/models/account_payment.py:30]: just nitpicking: could be nice to have the 2 fields next to each other in the .py file, with a comment explaining why we have 2 fields ;)

File: addons/account/models/account_payment.py (hunk 3 above)
Comment (qdp-odoo) [reply addons/account/models/account_payment.py:873]: ok, no big deal anyway

File: addons/hr_expense/models/account_payment.py (hunk 11 above)
Comment (qdp-odoo) [reply addons/hr_expense/models/account_payment.py:20]: yeah but the message was 
raise UserError(_("You cannot do this modification since the payment is linked to an expense report."))

Comment (william-andre) [reply addons/hr_expense/models/account_payment.py]: updated

File: addons/account/models/account_payment.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (william-andre) [reply addons/account/models/account_payment.py]: Done

File: addons/account/models/account_move.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (william-andre) [reply addons/account/models/account_move.py:53]: Yes, it's being removed AFAIK and it is easy to add a computed field afterwards anyway if needed. Not sure it is.

[the remark this answers is not in the document]
Comment (william-andre) [reply addons/account/models/account_move.py:53]: (And it's not exactly the same. The followup was only for customer invoices, whereas this is for both)

**Comment by william-andre** [comment]
> very detailed and motivated, even if the block "Implementation" would need a bit of a rework 

Are you telling me that the part that I copy-pasted from the task is alright, but the part that I wrote at 2am wasn't? :laughing:

File: addons/account/models/account_payment.py [hunk 15]
```diff
@@ -178,7 +192,8 @@ class AccountPayment(models.Model):
     amount_company_currency_signed = fields.Monetary(
         currency_field='company_currency_id', compute='_compute_amount_company_currency_signed', store=True)
     # used to get and display duplicate move warning if partner, amount and date match existing payments
-    duplicate_move_ids = fields.Many2many(comodel_name='account.move', compute='_compute_duplicate_move_ids')
+    duplicate_payment_ids = fields.Many2many(comodel_name='account.payment', compute='_compute_duplicate_move_ids')
```
Comment (MiquelRForgeFlow) [inline addons/account/models/account_payment.py:195]:
```suggestion
    duplicate_payment_ids = fields.Many2many(comodel_name='account.payment', compute='_compute_duplicate_payment_ids')
```
A bit of OCD... 😉

File: addons/account/models/account_payment.py (hunk 1 above)
Comment (qdp-odoo) [reply addons/account/models/account_payment.py:851]: mmkay... I don't have energy to fight on that anyway :sweat_smile:

**Comment by qdp-odoo** [comment]
> > very detailed and motivated, even if the block "Implementation" would need a bit of a rework
> 
> Are you telling me that the part that I copy-pasted from the task is alright, but the part that I wrote at 2am wasn't? 😆

hahaha dude :rofl: 

Don't you find it would be lame if it wasn't the case? I have a high opinion on our beloved PO :grin: 


@robodoo r+

File: addons/account/wizard/account_payment_register.py [hunk 16]
```diff
@@ -839,9 +865,9 @@ def _compute_qr_code(self):
     def _compute_duplicate_moves(self):
         for wizard in self:
             if wizard.can_edit_wizard:
-                wizard.duplicate_move_ids = self._fetch_duplicate_reference()
+                wizard.duplicate_payment_ids = self._fetch_duplicate_reference().get(0, self.env['account.payment'])
             else:
-                wizard.duplicate_move_ids = self.env['account.move']
+                wizard.duplicate_payment_ids = self.env['account.payment']
```
Comment (Levizar) [inline addons/account/wizard/account_payment_register.py:870]: Probably wasn't an issue isn't very likely to be one because it's a wizard but calling it on `self` inside a loop while the `_fetch_duplicate_reference` method below itself can't handle more than one record could one day in very specific condition be problematic.

It would be nice to correct that in another PR.

File: addons/sale/tests/test_payment_flow.py [hunk 17]
```diff
@@ -24,6 +24,10 @@ def setUpClass(cls):
         cls.currency = cls.sale_order.currency_id
         cls.partner = cls.sale_order.partner_invoice_id
 
+        cls.provider.journal_id.inbound_payment_method_line_ids.filtered(lambda l:
+            l.payment_provider_id == cls.provider
+        ).payment_account_id = cls.inbound_payment_method_line.payment_account_id
```
Comment (Levizar) [inline addons/sale/tests/test_payment_flow.py:29]: Isn't it boring to be forced to do that all the time in every test?
It also means that it won't be made automatically for the users anymore.

**Review by Levizar — COMMENTED** [review]
2 non important cents. :grin:

File: addons/sale/tests/test_payment_flow.py (hunk 17 above)
Comment (william-andre) [reply addons/sale/tests/test_payment_flow.py:29]: > Isn't it boring to be forced to do that all the time in every test?

Yes, but the point will be to not do it anymore in the future. For now I was mainly concentrating on not breaking anything.
In the future, we will be smarter on which methods require an account or not, and maybe create it automatically when needed

**Comment by qdp-odoo** [comment]
@robodoo r+

**Comment by qdp-odoo** [comment]
@robodoo r+

**Comment by qdp-odoo** [comment]
@robodoo r+

**Comment by jjscarafia** [comment]
I'm getting it wrong or now account.payment is more similar to a payment.transaction?
why not merge "sepa" world with something similar to payment.transaction?

I (we) really don't like loosing internal transfers. They are used a lot for:
* cash journals transfers (and/or combined with banks journals). On cash journals most users don't use statements. How would they encode de transfer?
* the moment where you decide to make a internal transfer (between banks and/or cash journals) is different than when the moves get into the bank (with good luck some minutes or hours) and then some more time till you reconcile. If I decide to make a transfer usually I already want to reflect it and then the reconciliation is separte thing on the bank journal.

I'm likely missing something, but the initial feedback from the AR community regarding this change hasn't been very positive.

I understand the motivation behind the change, but personally, I would have taken a different approach. Again, I might not have the full picture, but I wanted to share these concerns before the freeze, in case they help.

@william-andre

**Comment by william-andre** [comment]
@jjscarafia Merging with payment transactions was part of the initial spec, but it couldn't be done because of lack of time. We may or may not proceed with that more in the future, but we identified a few possible issues and it would have been a much deeper refactoring.

Regarding the internal transfer feature, we are aware of it, and the feature will be re-done before or right after the release of Odoo 18; maybe a bit differently though.

**Comment by jjscarafia** [comment]
Thanks a lot @william-andre !

Any chance to reconsider and leave the refactor for next major version?

Some more feeback (aditional to the internal transfers need)

**Modification of payment don't update journal entry. Bug?**
Is no problem not to have the journal entry being created on draft payment, but if a payment with outstanding setup creates a joural entry, going to draft and changing the payment should impact the journal entry. May be, when going back to draft the journal entry should be removed.
In summary, the removal of the sincronization is not a really big deal, but consistency should be kept when going back to draft, editing and posting again.

**Bugs?**
1. When the payment method has an outstanding account and a payment is made, it also registers a journal entry. However, if you try to reset the payment to draft, an user error arises: “A payment cannot have a journal entry if it is not confirmed” ..but there is no confirmed status in a payment now.
2. If the journal entry of a payment in process is deleted, the payment can not change its status and become blocked.
3. You can not send back to draft a payment that is in process. Is this in purpose?
4. lot of bugs in l10n_latam check (we'll follow it with l10n_ team)
5. add to batch not working (check image attached)

image

**Comment by jjscarafia** [comment]
Great! thanks @william-andre !

**Comment by Alsheh** [comment]
@william-andre I'm using version 18.0+e. When recording payments on an invoice, I can no longer determine if the invoice is partially or fully paid, as the "Amount Due" field no longer seems to reflect the outstanding balance accurately. How can we determine the remaining amount due?

**Comment by tde-banana-odoo** [comment]
> Multiple fixes are still ongoing, this is not a stable version; thanks for reporting it though! (note that everything was already known)

Now that the not stable version is stable, please note this broke appointment.

File: addons/account/tests/test_account_move_in_invoice.py [hunk 18]
```diff
@@ -1579,11 +1579,12 @@ def test_in_invoice_payment_register_wizard(self):
         self.assertFalse(move.payment_ids)  # don't auto reconcile payments
 
         # Reconcile manually, the move is now fully paid
-        (move.line_ids[-1] | payment.line_ids.filtered(lambda line: line.account_id == move.line_ids[-1].account_id)).reconcile()
+        (move.line_ids[-1] | payment.move_id.line_ids.filtered(lambda line: line.account_id == move.line_ids[-1].account_id)).reconcile()
 
         # If the move is already fully paid, we should alert the user
-        with self.assertRaisesRegex(UserError, r"You can only register payments for \(partially\) unpaid documents."):
-            move.action_force_register_payment()
+        with self.assertRaisesRegex(UserError, r"You can't register a payment because there is nothing left"):
```
[the remark this answers is not in the document]
Comment (hwangh95) [reply addons/account/tests/test_account_move_in_invoice.py:1585]: Sorry it's actually when accountant is not installed
